tests/testthat/test-model-compile.R

```diff
@@ -280,51 +331,9 @@ test_that("check_syntax() works", {
     mod_ok$check_syntax(stanc_options = list("allow-undefined", "warn-pedantic"), quiet = TRUE),
     regexp = NA
   )
-
-  pedantic_model <- write_stan_file("
```

Comment:
I removed the pedantic tests here since we have a separate test for pedantic mode below now